src/notification_handling.rs

```diff
@@ -240,6 +249,124 @@ fn handle_address_txs_confirmed_notification(
+        .handle()
+        .block_on(sdk.receive_payment(ReceivePaymentRequest {
+            amount_msat: data.amount_msat,
+            description: data.payer_comment.unwrap_or_default(),
+            preimage: None,
+            opening_fee_params: None,
+            use_description_hash: None,
+            expiry: None,
+            cltv: None,
+        }))
+        .map_to_runtime_error(
+            NotificationHandlingErrorCode::NodeUnavailable,
+            "Failed to create invoice",
+        )?;
+    if receive_payment_result.opening_fee_msat.is_some() {
+        runtime_error!(
+            NotificationHandlingErrorCode::InsufficientInboundLiquidity,
+            "Rejecting an inbound LNURL-pay payment because of insufficient inbound liquidity"
+        );
+    }
+
+    // Invoice is not persisted in invoices table because we are not interested in unpaid invoices
+    // resulting from incoming LNURL payments
+
+    // Store payment info (exchange rates, user preferences, etc...)
+    let user_preferences = UserPreferences {
+        fiat_currency: config.fiat_currency.clone(),
+        timezone_config: config.timezone_config.clone(),
+    };
+    let exchange_rate_provider =
+        ExchangeRateProviderImpl::new(environment.backend_url.clone(), Arc::new(auth));
+    let exchange_rates = exchange_rate_provider
+        .query_all_exchange_rates()
+        .map_to_runtime_error(
+            NotificationHandlingErrorCode::LipaServiceUnavailable,
+            "Failed to get exchange rates",
+        )?;
+
+    let db_path = format!("{}/{DB_FILENAME}", config.local_persistence_path);
+    let mut data_store = DataStore::new(&db_path)
+        .map_runtime_error_using(NotificationHandlingErrorCode::from_runtime_error)?;
+    data_store
+        .store_payment_info(
+            &receive_payment_result.ln_invoice.payment_hash,
+            user_preferences,
+            exchange_rates,
+            None,
+        )
+        .log_ignore_error(Level::Error, "Failed to persist payment info");
+
+    // Persist recipient information
+    data_store
+        .update_received_on(
+            &receive_payment_result.ln_invoice.payment_hash,
+            &data.recipient,
+        )
+        .log_ignore_error(Level::Error, "Failed to persist recipient info");
+
+    // Submit created invoice to backend
+    let async_auth = build_async_auth(&strong_typed_seed, &environment.backend_url)
+        .map_to_runtime_error(
+            NotificationHandlingErrorCode::LipaServiceUnavailable,
+            "Failed to authenticate against backend",
+        )?;
+    rt.handle()
+        .block_on(submit_lnurl_pay_invoice(
+            &environment.backend_url,
+            &async_auth,
+            data.id,
+            receive_payment_result.ln_invoice.bolt11,
+        ))
+        .map_runtime_error_to(NotificationHandlingErrorCode::LipaServiceUnavailable)?;
+
+    Ok(Notification::LnurlInvoiceCreated {
+        amount_sat: data.amount_msat / 1_000,
```

Comment:
We didn't have such field because some rounding needs to be done. I created such rounding methods in [ad9ddd3](https://github.com/getlipa/lipa-lightning-lib/pull/1050/commits/ad9ddd33ad1a56737ed4e83e8506b3e02f40ea67).